### PR TITLE
CI(pre-commit): Enable markdownlint fixes

### PR DESCRIPTION
**Pull request by echoix**
The pre-commit definition for markdownlint provides multiple hook ids, and one of them enabling to autofix some issues that markdownlint can fix.

This PR thus enables to auto-fix markdown files (like black and clang-format) for supported problems when running pre-commit locally. It is a quality of life improvement.

### CODE REVIEW

**Comment by echoix** [comment]
The unfixed files are fixed in https://github.com/OSGeo/grass/pull/3915, that is waiting to be reviewed and merged.

**Comment by echoix** [comment]
I updated the PR description, and this PR is awaiting an approval before merging.

**Review by neteler — APPROVED** [review]
Looks good.

**Pull request merged**
